dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -142,14 +142,14 @@ bool TarArchive::NextFile() {
   }
 
   const int64_t offset = stream_->Tell() + RoundToBlockSize(filesize_) - readoffset_;
+  current_header_ = offset;
   assert(offset >= 0);
   if (static_cast<size_t>(offset) >= stream_->Size()) {
     SetEof();
     return false;
   }
 
-  stream_->Seek(stream_->Tell() + RoundToBlockSize(filesize_) - readoffset_);
-  current_header_ = stream_->Tell();
+  stream_->Seek(offset);
```

Comment:
Actually, you should move `Seek` up, too.

Reply:
no, that's what this if statement is for. The stream will throw an error when we try to seek to the last element (at least the mapped one will)

Reply:
Which is... bullshit behavior. You can seek to the EOF in all implementations of a file I know of. Not being able to do so makes it impossible to seek to 0 in an empty file, which is ridiculous. We should fix the mapped file.

Reply:
I think it still an issue, right?

Reply:
Yes it is. And fixing it is _dirt cheap_.

Reply:
alright

Reply:
done